test/models/build_status_updater_test.exs

```diff
@@ -0,0 +1,25 @@
+defmodule Bigben.BuildStatusUpdaterTest do
+  use Bigben.ModelCase
+
+  alias Bigben.Branch
+  alias Bigben.Repo
+  alias Bigben.Build
+  alias Bigben.BuildStatusUpdater
+
+  import Bigben.Factory
+
+  test "get_or_create_branch creates a new branch when there is not one" do
+    branch = BuildStatusUpdater.get_or_create_branch(%{"branch" => "master"})
+
+    assert branch.name == "master"
+  end
+
+  test "get_branch finds existing by the same name" do
```

Comment:
This should also be `get_or_create_branch`. Also, I've had this issue as well, but I think this should be `get_or_*insert*_branch` to match Ecto terminology. It's really hard to break the habit of "create" when coming from Rails :P

Related: I'll be adding `insert`, `insert_pair` and `insert_list` to ExMachina to match Ecto's terminology: https://github.com/thoughtbot/ex_machina/issues/89